### PR TITLE
add Recommended design in dropdown component

### PR DESCRIPTION
**Pull request by patelvarun**

### CODE REVIEW

File: packages/base/Dropdown/index.js [hunk 1]
```diff
@@ -124,9 +124,42 @@ const Dropdown = ({ input, ...rest }) => {
         const overrideStyles = getOverrideStyles({ error, theme });
 
         const getOptionLabel = (option) => (
```
Comment (cadejscroggins) [inline packages/base/Dropdown/index.js:126]: Why do we only utilize this in the async dropdown? Should we also use it in the normal dropdown?

File: packages/base/Dropdown/index.js [hunk 2]
```diff
@@ -124,9 +124,42 @@ const Dropdown = ({ input, ...rest }) => {
...
+                alignItems: 'flex-start',
+                display: 'flex',
+                justifyContent: 'space-between',
+                mb: 1,
+              }}
+            >
+              <Box sx={{ flex: 'auto', mr: 2, wordBreak: 'break-all' }}>
+                {option.label}
+              </Box>
+              {option.isRecommended && (
```
Comment (cadejscroggins) [inline packages/base/Dropdown/index.js:139]: `isRecommended` seems really specific for a common module. Maybe we can rename this to `pill` or something similar and allow the content to be custom.

File: packages/base/Dropdown/index.js [hunk 3]
```diff
@@ -124,9 +124,42 @@ const Dropdown = ({ input, ...rest }) => {
...
+                <Box
+                  sx={{
+                    bg: 'orange',
+
```
Comment (cadejscroggins) [inline packages/base/Dropdown/index.js:143]: Can remove this whitespace.

File: packages/base/Dropdown/index.js (hunk 1 above)
Comment (patelvarun) [reply packages/base/Dropdown/index.js:126]: Yes, for now, it is only for async dropdown options. I will check how we can use with normal options

File: packages/base/Dropdown/index.js (hunk 3 above)
Comment (patelvarun) [reply packages/base/Dropdown/index.js:143]: Sure I am removing it.

File: packages/base/Dropdown/index.js (hunk 2 above)
Comment (patelvarun) [reply packages/base/Dropdown/index.js:139]: I will rename it to pill

**Pull request merged**